lib/tables.dart

```diff
@@ -81,7 +119,19 @@ class Table<T> {
       _doSort();
     }
 
-    _rebuildTable();
+    _scheduleRebuild();
+  }
+
+  void _scheduleRebuild() {
+    if (!hasPendingRebuild) {
+      // Set a flag to ensure we don't schedule rebuilds if there's already one
```

Comment:
Does some of this logic need to be pushed up to the `setState` implementation? Still call the closure, in order to update the model, but delay rebuilding the DOM until a timer fires?

We may just want a TODO: here to revisit when we want to spend a bit more time making the framework more efficient.

Reply:
Yeah, I think we need to do something but I wasn't sure the best way (we can't just have a single flag, because `setState` can be called with different functions). Maybe we need a queue of functions that get added to, then when RAF fires, we execute them sequentially and clear the queue. I'll add a TODO about this.

Reply:
Actually, that won't work - this flag is specifically to *skip* calls if there is already one queued. This is only valid if the function being queued always brings things up-to-date. So if we push it into `setState`, we may need some way of providing an identifier to group together calls that are allowed to be squashed together in that way?

Reply:
Yeah, thinking about this, your right. This would need a bunch more knowledge to be able to batch process things in setState(). Probably don't need a TODO: here, but we may want to keep an eye out for common patterns as we built our more of the app.

Reply:
What about if we had a mixin that specifically is for rebuilds tied to RAF? This seems like the most common case and by having queueing a rebuild method that takes no args we know it's safe to drop additional requests while there's still one pending:

```dart
class FrameSyncedRebuildMixin {
  bool _hasPendingRebuild = false;
  void scheduleRebuild() {
    if (!_hasPendingRebuild) {
      // Set a flag to ensure we don't schedule rebuilds if there's already one
      // in the queue.
      _hasPendingRebuild = true;
      window.requestAnimationFrame((_) {
        _hasPendingRebuild = false;
        rebuild();
      });
    }
  }

  void rebuild() {}
}
```

With this mixin, you override `rebuild` and then just call `scheduleRebuild` whenever you want (and as often as you like).